packages/dappmanager/src/modules/stakerConfig/setStakerConfig.ts

```diff
@@ -508,3 +520,70 @@ function setStakerConfigOnDb<T extends Network>({
       throw new Error(`Unsupported network: ${network}`);
   }
 }
+
+function checkLodestarMinVersions(
+  lodestarDnpName: string,
+  selectedExecClient: StakerItemOk<Network, "execution"> | undefined,
+  isSignerSelected: boolean | undefined,
+  web3signerPkg: InstalledPackageDataApiReturn | undefined
+): void {
+  if (!selectedExecClient) {
+    throw Error(`Execution client is not selected.`);
+  }
+
+  const lodestarMinVersions =
+    params.lodestarStakersMinimumVersions[lodestarDnpName];
+
+  if (!lodestarMinVersions) {
+    throw Error(
+      `Lodestar ${lodestarDnpName} minimum versions are not defined.`
+    );
+  }
+
+  const selectedExecClientVersion = selectedExecClient.data?.metadata.version;
+
+  //Throw error if execution client version cannot be fetched
+  if (selectedExecClientVersion === undefined) {
+    throw Error(
+      `Execution client ${selectedExecClient.dnpName} version could not be fetched.`
+    );
+  }
+
+  checkStakersItemLodestarCompatibility({
+    version: selectedExecClientVersion,
+    minVersion: lodestarMinVersions[selectedExecClient.dnpName],
+    dnpName: selectedExecClient.dnpName
+  });
+
+  if (!isSignerSelected || !web3signerPkg) return;
+
+  const selectedSignerVersion = web3signerPkg.version;
+
+  if (!selectedSignerVersion) {
+    throw Error(
+      `Signer ${web3signerPkg.dnpName} version could not be fetched.`
+    );
+  }
+
+  checkStakersItemLodestarCompatibility({
+    version: selectedSignerVersion,
+    minVersion: lodestarMinVersions[web3signerPkg.dnpName],
+    dnpName: web3signerPkg.dnpName
+  });
+}
+
+function checkStakersItemLodestarCompatibility({
```

Comment:
I do not see the point of having a function that contains on it another function. You should use the `semver.lt` directly where required